Survey/Survey/Sources/Domain/Entities/LoginToken.swift

```diff
@@ -0,0 +1,17 @@
+//
+//  LoginToken.swift
+//  Survey
+//
+//  Created by Taher on 4/11/22.
+//
+
+import Foundation
+
+protocol LoginToken {
+
+    var accessToken: String { get }
+    var tokenType: String { get }
+    var expiresIn: Int { get }
+    var refreshToken: String { get }
+    var createdAt: Int { get }
+}
```

Comment:
Although the API's response is `Int` for `expiresIn` and `createdAt`, consider make it to `Date` (or `Date?` if you want a little peace of mind). In that case, you might want to set your `JSONDecoder`'s `dateDecodingStrategy` to `.secondsSince1970`

Reply:
updated in this [commit](https://github.com/Shayokh144/ic-survey-swiftui-combine/pull/43/commits/08e91693e844c6ce389503dced572cfcc5f17cea)